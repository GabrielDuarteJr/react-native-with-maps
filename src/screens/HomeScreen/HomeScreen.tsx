import React, {useContext} from 'react';
import {CarContext} from '~/contexts/CarContext';
import NavigationMapComponent from '~/components/NavigationMapComponent/NavigationMapComponent';
import CloseButtonComponent from '~/components/CloseButtonComponent/CloseButtonComponent';
import InfoViewComponent from '~/components/InfoViewComponent/InfoViewComponent';
import DestinationMock from '~/utils/mocks/DestinationMock';
import {Container} from './styles';

const HomeScreen: React.FunctionComponent = () => {
  const {origin, carOrientation, carDistance, carTime, onRestart} =
    useContext(CarContext);

  return (
    <Container>
      <NavigationMapComponent
        origin={origin}
        destination={DestinationMock}
        carOrientation={carOrientation}
      />
      <CloseButtonComponent onClose={onRestart} />
      <InfoViewComponent
        onCancel={onRestart}
        distance={carDistance}
        time={carTime}
      />
    </Container>
  );
};

export default HomeScreen;
