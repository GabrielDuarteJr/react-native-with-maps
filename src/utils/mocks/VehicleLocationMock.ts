interface VehicleLocationInterface {
  cordinates: Cordinates;
  orientation: number;
  distance: number;
  time: string;
}

const VehicleLocationMock: Array<VehicleLocationInterface> = [
  {
    cordinates: {
      latitude: -18.914931,
      longitude: -48.244491,
    },
    orientation: -20,
    distance: 6.5,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.915202,
      longitude: -48.244386,
    },
    orientation: -110,
    distance: 6.4,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.915261,
      longitude: -48.244156,
    },
    orientation: -110,
    distance: 6.3,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.915306,
      longitude: -48.24399,
    },
    orientation: -110,
    distance: 6.2,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.915351,
      longitude: -48.243829,
    },
    orientation: -110,
    distance: 6.1,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.915404,
      longitude: -48.243645,
    },
    orientation: -110,
    distance: 6.0,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.915439,
      longitude: -48.243513,
    },
    orientation: -110,
    distance: 5.9,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.91548,
      longitude: -48.243373,
    },
    orientation: -110,
    distance: 5.8,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.915514,
      longitude: -48.243241,
    },
    orientation: -20,
    distance: 5.7,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.915629,
      longitude: -48.243253,
    },
    orientation: -20,
    distance: 5.6,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.915719,
      longitude: -48.243285,
    },
    orientation: -20,
    distance: 5.5,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.915842,
      longitude: -48.243332,
    },
    orientation: -20,
    distance: 5.4,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.915973,
      longitude: -48.243372,
    },
    orientation: -20,
    distance: 5.3,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.915973,
      longitude: -48.243372,
    },
    orientation: -20,
    distance: 5.2,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.916049,
      longitude: -48.243392,
    },
    orientation: -110,
    distance: 5.1,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.916088,
      longitude: -48.243288,
    },
    orientation: -110,
    distance: 5.0,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.91612,
      longitude: -48.243143,
    },
    orientation: -110,
    distance: 4.9,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.916167,
      longitude: -48.242993,
    },
    orientation: -110,
    distance: 4.8,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.916215,
      longitude: -48.242817,
    },
    orientation: -110,
    distance: 4.7,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.916247,
      longitude: -48.242677,
    },
    orientation: -110,
    distance: 4.6,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.916275,
      longitude: -48.242543,
    },
    orientation: -110,
    distance: 4.5,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.916322,
      longitude: -48.242381,
    },
    orientation: -110,
    distance: 4.4,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.916369,
      longitude: -48.242151,
    },
    orientation: -110,
    distance: 4.3,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.916397,
      longitude: -48.241999,
    },
    orientation: -110,
    distance: 4.2,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.916451,
      longitude: -48.241831,
    },
    orientation: -110,
    distance: 4.1,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.916498,
      longitude: -48.241657,
    },
    orientation: -110,
    distance: 4.0,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.916555,
      longitude: -48.24144,
    },
    orientation: -110,
    distance: 3.9,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.916589,
      longitude: -48.241317,
    },
    orientation: -110,
    distance: 3.8,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.916646,
      longitude: -48.24108,
    },
    orientation: -110,
    distance: 3.7,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.916682,
      longitude: -48.240917,
    },
    orientation: -110,
    distance: 3.6,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.916715,
      longitude: -48.240777,
    },
    orientation: -110,
    distance: 3.5,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.916734,
      longitude: -48.240678,
    },
    orientation: -20,
    distance: 3.4,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.916819,
      longitude: -48.240704,
    },
    orientation: -20,
    distance: 3.3,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.916958,
      longitude: -48.240747,
    },
    orientation: -20,
    distance: 3.2,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.916959,
      longitude: -48.240745,
    },
    orientation: -20,
    distance: 3.1,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.917252,
      longitude: -48.240832,
    },
    orientation: -20,
    distance: 3.0,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.917324,
      longitude: -48.240849,
    },
    orientation: -20,
    distance: 2.9,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.917477,
      longitude: -48.240897,
    },
    orientation: -20,
    distance: 2.8,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.917598,
      longitude: -48.240929,
    },
    orientation: -20,
    distance: 2.7,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.91769,
      longitude: -48.240956,
    },
    orientation: -20,
    distance: 2.6,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.917756,
      longitude: -48.240976,
    },
    orientation: -20,
    distance: 2.5,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.917852,
      longitude: -48.241005,
    },
    orientation: -20,
    distance: 2.4,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.917988,
      longitude: -48.241044,
    },
    orientation: -20,
    distance: 2.3,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.918167,
      longitude: -48.241094,
    },
    orientation: -20,
    distance: 2.2,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.918271,
      longitude: -48.241121,
    },
    orientation: -20,
    distance: 2.1,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.918365,
      longitude: -48.241147,
    },
    orientation: -20,
    distance: 2.0,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.918513,
      longitude: -48.241191,
    },
    orientation: -20,
    distance: 1.9,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.918557,
      longitude: -48.241201,
    },
    orientation: -20,
    distance: 1.8,
    time: '2 min',
  },
  {
    cordinates: {
      latitude: -18.918653,
      longitude: -48.241233,
    },
    orientation: -20,
    distance: 1.7,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.918839,
      longitude: -48.241277,
    },
    orientation: -20,
    distance: 1.6,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.91888,
      longitude: -48.241291,
    },
    orientation: -20,
    distance: 1.5,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.918943,
      longitude: -48.241307,
    },
    orientation: -20,
    distance: 1.4,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.918967,
      longitude: -48.241317,
    },
    orientation: -20,
    distance: 1.3,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.919096,
      longitude: -48.241352,
    },
    orientation: -20,
    distance: 1.2,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.919156,
      longitude: -48.241367,
    },
    orientation: -20,
    distance: 1.0,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.919296,
      longitude: -48.241404,
    },
    orientation: -20,
    distance: 0.9,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.919425,
      longitude: -48.241444,
    },
    orientation: -20,
    distance: 0.8,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.919474,
      longitude: -48.241457,
    },
    orientation: -20,
    distance: 0.7,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.919569,
      longitude: -48.241487,
    },
    orientation: -20,
    distance: 0.6,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.919804,
      longitude: -48.241546,
    },
    orientation: -20,
    distance: 0.5,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.919905,
      longitude: -48.241574,
    },
    orientation: -20,
    distance: 0.4,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.920058,
      longitude: -48.241612,
    },
    orientation: -20,
    distance: 0.3,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.920189,
      longitude: -48.241814,
    },
    orientation: 70,
    distance: 0.2,
    time: '1 min',
  },
  {
    cordinates: {
      latitude: -18.920153,
      longitude: -48.241946,
    },
    orientation: 70,
    distance: 0.1,
    time: '1 min',
  },
];

export default VehicleLocationMock;
