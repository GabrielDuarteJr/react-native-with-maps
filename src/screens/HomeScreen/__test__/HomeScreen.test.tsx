import 'react-native';
import React from 'react';
import {render, screen, cleanup} from '@testing-library/react-native';
import VehicleLocationMock from '~/utils/mocks/VehicleLocationMock';
import HomeScreen from '../HomeScreen';

jest.mock('react-native-maps', () => {
  const {View, Text} = require('react-native');
  const MockMapView = (props: any) => {
    return <View>{props.children}</View>;
  };
  const MockMarker = (props: any) => {
    return (
      <View>
        <Text>{props.coordinate.toString()}</Text>
        <Text>{props.rotation}</Text>
        {props.children}
      </View>
    );
  };
  return {
    __esModule: true,
    default: MockMapView,
    Marker: MockMarker,
  };
});

jest.mock('react-native-maps-directions', () => {
  const {View} = require('react-native');
  const MockMapDirectionsView = (props: any) => {
    return <View>{props.children}</View>;
  };
  return {
    __esModule: true,
    default: MockMapDirectionsView,
  };
});

afterEach(cleanup);

describe('HomeScreen', () => {
  it('should render without crashing', () => {
    const tree = render(<HomeScreen />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should display the props correctly', () => {
    render(<HomeScreen />);

    const initialPropCordinates = screen.getAllByText(
      VehicleLocationMock[0].cordinates.toString(),
    );
    const initialPropOrientation = screen.getAllByText(
      VehicleLocationMock[0].orientation.toString(),
    );
    const initialPropDistance = screen.getAllByText(
      `${VehicleLocationMock[0].distance.toString()} km`,
    );
    const initialPropTime = screen.getAllByText(
      VehicleLocationMock[0].time.toString(),
    );

    expect(initialPropCordinates[0]).toBeDefined();
    expect(initialPropOrientation[0]).toBeDefined();
    expect(initialPropDistance[0]).toBeDefined();
    expect(initialPropTime[0]).toBeDefined();
  });
});
