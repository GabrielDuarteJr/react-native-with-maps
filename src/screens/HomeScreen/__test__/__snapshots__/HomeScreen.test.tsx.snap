// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should render without crashing 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View>
    <View />
    <View>
      <Text>
        [object Object]
      </Text>
      <Text>
        -20
      </Text>
      <SvgMock />
    </View>
    <View>
      <Text>
        [object Object]
      </Text>
      <Text />
      <SvgMock />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": 50,
          "justifyContent": "center",
          "paddingHorizontal": "5%",
          "position": "absolute",
          "top": "5%",
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 30,
          "elevation": 1,
          "height": 44,
          "justifyContent": "center",
          "opacity": 1,
          "width": 44,
        }
      }
      testID="button-container"
    >
      <SvgMock />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "bottom": 8,
          "height": "35%",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 8,
            "elevation": 2,
            "height": "100%",
            "width": "90%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F7F7F7",
              "borderBottomColor": "#EFEFF4",
              "borderBottomWidth": 1,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "display": "flex",
              "flexDirection": "row",
              "height": "25%",
              "paddingHorizontal": 10,
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": "100%",
                "justifyContent": "center",
                "width": "20%",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/Profile.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 50,
                  "width": 50,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "height": "100%",
                "justifyContent": "center",
                "width": "40%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#242E42",
                  "fontFamily": "SF UI Display",
                  "fontSize": 17,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Gregory Smith
          </Text>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
              ]
            }
          >
            <SvgMock />
            <Text
              style={
                Array [
                  Object {
                    "color": "#C8C7CC",
                    "fontFamily": "SF UI Display",
                    "fontSize": 15,
                    "lineHeight": 18,
                    "marginLeft": 5,
                  },
                ]
              }
            >
              4.9
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-around",
                "paddingLeft": 15,
                "width": "40%",
              },
            ]
          }
        >
          <SvgMock />
          <SvgMock />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#EFEFF4",
              "borderBottomWidth": 1,
              "display": "flex",
              "flexDirection": "row",
              "height": "35%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": "100%",
                "justifyContent": "center",
                "width": "20%",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#4CE5B1",
                  "borderRadius": 10,
                  "borderWidth": 2,
                  "height": 19,
                  "justifyContent": "center",
                  "width": 19,
                },
              ]
            }
          >
            <SvgMock />
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#C8C7CC",
                  "borderStyle": "dashed",
                  "borderWidth": 1,
                  "height": 25,
                },
              ]
            }
          />
          <SvgMock />
        </View>
        <View
          style={
            Array [
              Object {
                "height": "100%",
                "width": "80%",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "#EFEFF4",
                  "borderBottomWidth": 1,
                  "height": "50%",
                  "justifyContent": "center",
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#242E42",
                    "fontFamily": "SF UI Display",
                    "fontSize": 16,
                    "lineHeight": 20,
                  },
                ]
              }
            >
              7958 Swift Village
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "height": "50%",
                  "justifyContent": "center",
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#242E42",
                    "fontFamily": "SF UI Display",
                    "fontSize": 16,
                    "lineHeight": 20,
                  },
                ]
              }
            >
              105 William St, Chicago, US
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": "40%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "height": "50%",
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "25%",
                },
              ]
            }
          >
            <SvgMock />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "25%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#C8C7CC",
                    "fontFamily": "SF UI Display",
                    "fontSize": 13,
                    "lineHeight": 16,
                    "marginBottom": 5,
                  },
                ]
              }
            >
              Distance
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#242E42",
                    "fontFamily": "SF UI Display",
                    "fontSize": 15,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              6.5 km
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "25%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#C8C7CC",
                    "fontFamily": "SF UI Display",
                    "fontSize": 13,
                    "lineHeight": 16,
                    "marginBottom": 5,
                  },
                ]
              }
            >
              Time
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#242E42",
                    "fontFamily": "SF UI Display",
                    "fontSize": 15,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              2 min
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "25%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#C8C7CC",
                    "fontFamily": "SF UI Display",
                    "fontSize": 13,
                    "lineHeight": 16,
                    "marginBottom": 5,
                  },
                ]
              }
            >
              Price
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#242E42",
                    "fontFamily": "SF UI Display",
                    "fontSize": 15,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              $25.00
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": "50%",
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#242E42",
                "borderRadius": 8,
                "height": "80%",
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
            testID="button-cancel-container"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "SF UI Display",
                    "fontSize": 17,
                    "lineHeight": 20,
                  },
                ]
              }
            >
              Cancel Request
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
